### PR TITLE
8250563: Add KVHashtable::add_if_absent

### PR DESCRIPTION
**Pull request by iklam**
Please review this XS change. I added a new **KVHashtable::add_if_absent** function (modeled after ResourceHashtable::put_if_absent from JDK-8244733).

- I used "add" instead of "put" to be consistent with the naming convention in utility/hashtable.hpp
- I also fixed a type in the comments in resourceHashtable.hpp

Running mach5 tiers1/2.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8250563](https://bugs.openjdk.java.net/browse/JDK-8250563): Add KVHashtable::add_if_absent

### Reviewers
 * [Calvin Cheung](https://openjdk.java.net/census#ccheung) (@calvinccheung - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/66/head:pull/66`
`$ git checkout pull/66`

### CODE REVIEW

File: src/hotspot/share/utilities/hashtable.hpp [hunk 1]
```diff
@@ -319,6 +319,27 @@ class KVHashtable : public BasicHashtable<F> {
     return NULL;
   }
 
+  // Look up the key.
+  // If an entry for the key exists, leave map unchanged and return a pointer to its value.
+  // If no entry for the key exists, create a new entry from key and value and return a
+  //  pointer to the value.
+  // *p_created is true if entry was created, false if entry pre-existed.
+  V* add_if_absent(K const& key, V const& value, bool* p_created) {
```
Comment (shipilev) [inline src/hotspot/share/utilities/hashtable.hpp:327]: Does it really need `const &` here? It looks inconsistent with `add(K,V)` and `lookup(K)` in the same class.

**Review by shipilev — CHANGES_REQUESTED** [review]
Otherwise looks good, modulo the single signature question in review.

File: src/hotspot/share/utilities/hashtable.hpp (hunk 1 above)
Comment (iklam) [reply src/hotspot/share/utilities/hashtable.hpp:327]: I'll remove the `const&`. I copied that from resourceHash.hpp but it doesn't fit in with the rest of hashtable.hpp.

**Comment by calvinccheung** [comment]
Looks good.

**Review by calvinccheung — APPROVED** [review]

**Review by coleenp — APPROVED** [review]

**Pull request closed without merging**